ci-operator/step-registry/cucushift/ota/preupgrade/cucushift-ota-preupgrade-commands.sh

```diff
@@ -268,6 +281,103 @@ function pre-OCP-24358(){
+            break
+        fi
+        sleep 1m
+        retry=$((retry-1))
+    done
+    if [[ "${reason}" != *"FeatureGates_RestrictedFeatureGates_TechPreviewNoUpgrade"* ]]; then
+        echo "Error: 32747 After waiting 10 minutes, FeatureGates_RestrictedFeatureGates_TechPreviewNoUpgrade still not appears"
+        return 1
+    fi
+
+    retry=5
+    while (( retry > 0 )); do
+        alerts=$( get_prometheus_api "v1/alerts" )
+        alert="$(echo "${alerts}" | yq  -r '.data.alerts[]| select(.labels.alertname == "ClusterNotUpgradeable")')"
+        namespace="$(echo "${alert}" | yq -r ".labels.namespace")"
+        severity="$(echo "${alert}" | yq -r ".labels.severity")"
+        description="$(echo "${alert}" | yq -r ".annotations.description")"
+        state="$(echo "${alert}" | yq -r ".state")"
+        if [[ "${namespace}" != "openshift-cluster-version" ]] \
+            || [[ "${severity}" != "info" ]] \
+            || [[ "${description}" != *"In most cases, you will still be able to apply patch releases."* ]] \
+            || [[ "${state}" != "pending" ]]; then
+            sleep 1m
+            retry=$((retry-1))
+            continue
+        fi
+        break
+    done
+    echo -e "Alert ClusterNotUpgradeable at beginning:\n${alert}"
+    if [[ "${namespace}" != "openshift-cluster-version" ]]; then
+        echo "Error: 32747 namespace is incorrect, expected is openshift-cluster-version, but observed is ${namespace}"
+        return 1
+    fi
+    if [[ "${severity}" != "info" ]]; then
+        echo "Error: 32747 severity is incorrect, expected is info, but observed is ${severity}"
+        return 1
+    fi
+    if [[ "${description}" != *"In most cases, you will still be able to apply patch releases."* ]]; then
+        echo "Error: 32747 description is incorrect, expected is 'In most cases, you will still be able to apply patch releases.', but observed is '${description}'"
+        return 1
+    fi
+    if [[ "${state}" != "pending" ]]; then
+        echo "Error: 32747 state is incorrect, expected is pending, but observed is ${state}"
+        return 1
+    fi
+    echo "Alert ClusterNotUpgradeable at beginning is correct"
+
+    retry=0
+    while (( retry < 90 ));do
+        alerts=$( get_prometheus_api "v1/alerts" )
+        alert="$(echo "${alerts}" | yq  -r '.data.alerts[]| select(.labels.alertname == "ClusterNotUpgradeable")')"
+        state="$(echo "${alert}" | yq -r ".state")"
+        if [[ "${state}" == "firing" ]]; then
+            if [[ ${retry} -lt 60 ]]; then
+                echo "Error: 32747 Alerts should be changed for at least 1 hour: https://github.com/openshift/cluster-version-operator/blob/8a8bca5df3bd89f8caab7c185f407f6a6e2697c8/install/0000_90_cluster-version-operator_02_servicemonitor.yaml#L87-L93"
+                return 1
+            fi
+            break
+        fi
+        echo "Attempted ${retry}"
+        sleep 1m
+        retry=$((retry+1))
+    done
+    if [ -z "${alert}" ] ; then
```

Comment:
This is not necessary? If `alert` is null, above test will fail at line https://github.com/openshift/release/pull/60108/files#diff-c587fa8e293fdade6e68f947f6b3b4c7f27c3967a4ad98c6c7409f3a3d50ef6dR343

Reply:
the if condition should be:
```
if [[ "${state}" != "firing" ]]; then
```